### PR TITLE
add exponential backoff for ethereum tx receipt querying

### PR DESCRIPTION
**Pull request by juniuszhou**

### CODE REVIEW

**Review by jzvikart — APPROVED** [review]

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 1]
```diff
@@ -202,13 +202,16 @@ func RelayBatchProphecyCompletedToEthereum(
...
+	maxRetries := 5
 	i := 0
+	sleepDuration := time.Duration(2)
 
 	// if there is an error getting the tx, or if the tx fails, retry 60 times
 	for i < maxRetries {
-		// sleep 2 seconds to wait for tx to go through before querying.
-		sleepThread(2)
+		// sleep to wait for tx to go through before querying.
+		sleepThread(sleepDuration)
```
Comment (smartyalgo) [inline cmd/ebrelayer/txs/relayToEthereum.go:214]: Move this to L226 so it doesnt sleep for the first time

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 2]
```diff
@@ -217,10 +220,11 @@ func RelayBatchProphecyCompletedToEthereum(
 
 		if err != nil {
 			sugaredLogger.Errorw("Failed to submit to ethereum client", "error", err)
-			sleepThread(1)
 		} else {
 			break
 		}
+		// exponential backoff
+		sleepDuration *= sleepDuration
```
Comment (smartyalgo) [inline cmd/ebrelayer/txs/relayToEthereum.go:227]: I dont think this is the right implementation

```
i | sleepDuration value at end of iteration
0 | 2 * 2 = 4
1 | 4 * 4 = 16
2 | 16 * 16 = 256 
3 | 256 * 256 = 65536 = 1092 minutes
4 | 65536 * 63356 = 49710 days

Comment (smartyalgo) [reply cmd/ebrelayer/txs/relayToEthereum.go:227]: I think you want 
```
sleepDuration *= 2

Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToEthereum.go:227]: will set maxRetries as 4, last sleep about 4 min, then 
i | sleep | sleepDuration after sleep
0 | 2 | 4
1 | 4 | 16
2 | 16 | 256
3 | 256 | 65536
4 exit from loop

**Review by Brando753 — APPROVED** [review]
looks good

**Pull request merged**